hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/ZookeeperWorkloadConstraintsRepository.java

```diff
@@ -0,0 +1,57 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload.constraints;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.curator.framework.CuratorFramework;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.api.TopicName;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperBasedRepository;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperPaths;
+
+import java.util.HashMap;
+
+import static org.apache.commons.lang.ArrayUtils.isNotEmpty;
+
+public class ZookeeperWorkloadConstraintsRepository extends ZookeeperBasedRepository implements WorkloadConstraintsRepository {
+
+    private static final Logger logger = LoggerFactory.getLogger(ZookeeperWorkloadConstraintsRepository.class);
+
+    public ZookeeperWorkloadConstraintsRepository(CuratorFramework zookeeper, ObjectMapper mapper, ZookeeperPaths paths) {
+        super(zookeeper, mapper, paths);
+    }
+
+    @Override
+    public ConsumersWorkloadConstraints getConsumersWorkloadConstraints() {
+        ConsumersWorkloadConstraints workloadConstraints = new ConsumersWorkloadConstraints(new HashMap<>(), new HashMap<>());
```

Comment:
How about, instead of creating an instance of this class and mutating its children, create two hashmaps, feed them with values, and in the end create an instance of `ConsumersWorkloadConstraints`. 
This way you can create an immutable object.